lua/lvs_framework/autorun/lvs_bulletsystem.lua

```diff
@@ -199,9 +199,24 @@ if SERVER then
 
 		setmetatable( bullet, NewBullet )
 
+		--[[
+			Forgive the lack of optimisation here
+			Using cross products caused spread to shrink awkwardly as the shot direction becomes parallel with worldspace z axis.
+			Using a simplified '(bullet.Dir = 2 * data.Dir + randomOnUnitSphere() * data.Spread):GetNormalized() will not work here because data.Spread.z is always zero for 'some' reason.
```

Comment:
its always 0 because i messed it up in the templates/presets. See

https://github.com/SpaxscE/lvs_planes/blob/main/lua/lvs_framework/autorun/sh_plane_defaultweapons.lua#L23

The original system needs Y and Z, X is unused iirc. Reason for spread being off in the original system is because of the thing mentioned above. Nothing really wrong here

Reply:
Assuming you fix your template and add the Z coordinate, it is possible to bring back the old bullet spread code and everything should work just fine.

The issue that remains is that VectorRand() generates a random point within a unit cube, which in some cases causes the euclidean distance d=√x²+y²+z² of the randomly generated vector to exceed the length of 1.0 which turns the bullet spread from a circle into a rectangle, this can be a problem, because a rectangle of same size has a more surface area than a circle, causing all weapons to be about ~33% less accurate than they would be with a proper circular spread.

Thankfully this issue can be fixed very easily, by just replacing VectorRand() with a fuction that generates a random point on a unit sphere instead. Do not try to scale or normalize the result of VectorRand, conforming a point cube into a sphere causes weird '+' shaped pattern in the bullet spread.